zebra-state/src/error.rs

```diff
@@ -95,6 +97,19 @@ pub enum ValidateContextError {
     #[non_exhaustive]
     EarlyTransparentSpend { outpoint: transparent::OutPoint },
 
+    #[error("unshielded transparent coinbase spend: {outpoint:?} must be spent in a transaction with all shielded outputs")]
+    #[non_exhaustive]
+    UnshieldedTransparentCoinbaseSpend { outpoint: transparent::OutPoint },
+
+    #[error("immature transparent coinbase spend: attempt to spend {outpoint:?} at {spend_height:?}, but it must be spent after {min_spend_height:?}, at least {MIN_TRANSPARENT_COINBASE_MATURITY:?} blocks after it was created {created_height:?}")]
```

Comment:
Nit: Would it make sense to start breaking the long error messages in multi-line strings?

```suggestion
    #[error(
        "immature transparent coinbase spend: attempt to spend {outpoint:?} at {spend_height:?}, \
        but it must be spent after {min_spend_height:?}, \
        at least {MIN_TRANSPARENT_COINBASE_MATURITY:?} blocks \
        after it was created {created_height:?}"
    )]
```